dlt/destinations/athena/athena.py

```diff
@@ -0,0 +1,261 @@
+
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, schema: Schema, config: AthenaClientConfiguration) -> None:
+        sql_client = AthenaSQLClient(
+            self.make_dataset_name(schema, config.dataset_name, config.default_schema_name),
+            config
+        )
+        super().__init__(schema, config, sql_client)
+
+    def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
+        # never truncate tables in athena
+        super().initialize_storage([])
+
+    def _get_column_def_sql(self, c: TColumnSchema) -> str:
+        return f"`{c['name']}` {self._to_db_type(c['data_type'])}"
+
+    @classmethod
+    def _to_db_type(cls, sc_t: TDataType) -> str:
+        if sc_t == "wei":
+            return SCT_TO_HIVET["decimal"] % cls.capabilities.wei_precision
+        if sc_t == "decimal":
+            return SCT_TO_HIVET["decimal"] % cls.capabilities.decimal_precision
+        return SCT_TO_HIVET[sc_t]
+
+    @classmethod
+    def _from_db_type(cls, hive_t: str, precision: Optional[int], scale: Optional[int]) -> TDataType:
+        for key, val in HIVET_TO_SCT.items():
+            if hive_t.startswith(key):
+                return val
+        return None
+
+    def get_schema_by_hash(self, version_hash: str) -> StorageSchemaInfo:
+        query = f"""SELECT {self.VERSION_TABLE_SCHEMA_COLUMNS} FROM "{VERSION_TABLE_NAME}" WHERE version_hash = '{version_hash}';"""
+        try:
+            return self._row_to_schema_info(query)
+        except OperationalError:
+            return None
+
+    def _get_table_update_sql(self, table_name: str, new_columns: Sequence[TColumnSchema], generate_alter: bool) -> List[str]:
+        # TODO: get bucket var from outside
+        bucket = self.config.staging_config.bucket_url
+        dataset = self.sql_client.dataset_name
+        schema_name = self.schema.name
+        sql: List[str] = []
+
+        # for the system tables we need to create empty iceberg tables to be able to run, DELETE and UPDATE queries
+        is_iceberg = self.schema.tables[table_name].get("write_disposition", None) == "skip"
+        columns = ", ".join([self._get_column_def_sql(c) for c in new_columns])
+
+        # this will fail if the table prefix is not properly defined
+        table_prefix = path_utils.get_table_prefix(self.config.staging_config.layout, schema_name=schema_name, table_name=table_name)
+        location = f"{bucket}/{dataset}/{table_prefix}"
+
+        if is_iceberg and not generate_alter:
+            sql.append(f"""CREATE TABLE `{table_name}`
+                    ({columns})
+                    LOCATION '{location}'
+                    TBLPROPERTIES ('table_type'='ICEBERG', 'format'='parquet');""")
+        elif not generate_alter:
+            sql.append(f"""CREATE EXTERNAL TABLE `{table_name}`
+                    ({columns})
+                    STORED AS PARQUET
+                    LOCATION '{location}';""")
```

Comment:
as stated above, you want to give the possibility to create partitioned tables. In this specific case, for external table there is something more to do as partition values need to be added after writing to s3 is done.